### DISCUSSION TITLE
XMPP URI

### DISCUSSION THREAD

**Discussion by Neustradamus** [community/community#23192] [New to GitHub]
<p>Hello Team,</p>
<p>Can you see for add XMPP URI support in <a href="https://github.com/" rel="nofollow noopener">https://github.com/</a> and in this community too?</p>
<p>XMPP URI support must be added because there actually a problem, it is detected “mailto”.</p>
<p>Examples:</p>
<ul>
<li>xmpp:username@domain.tld</li>
<li>xmpp:username@domain.tld?message</li>
<li>xmpp:mucroom@domain.tld?join</li>
</ul>
<p>RFC5122: <a href="https://tools.ietf.org/html/rfc5122" rel="nofollow noopener">https://tools.ietf.org/html/rfc5122</a></p>
<p>Can you update it?</p>
<p>Thanks in advance.</p>

**Comment by darrowby385** [comment, marked as answer]
<p>Hi, <a class="mention" href="/neustradamus">@neustradamus</a>!</p>
<p>You’re correct that we don’t currently support rendering clickable links for XMPP URIs, both here in our Community Forum, and on GitHub itself. It’s also true that the xmpp: component of a URI is ignored, leading the rest to be detected as an email address, and rendered as a mailto: link. This latter effect is not so much a bug as a flow-on consequence of not supporting detection and rendering of XMPP URIs in the first place.</p>
<p>I can definitely appreciate the value of improving how we render XMPP URIs, so thank you for raising this! I can’t make any promises at to if or when this might change, but I’ve lodged a feature request internally for our engineering and product teams to consider. We don’t comment publicly on our feature development pipeline, but you can keep abreast of the latest updates by monitoring our changelog at <a href="https://github.blog/changelog/" rel="nofollow noopener">https://github.blog/changelog/</a></p>
<p>I hope that helps!</p>
<p>Friday</p>

**Comment by Neustradamus** [comment]
<p>Thanks <a class="mention" href="/darrowby385">@darrowby385</a> for your reply!</p>

**Comment by Neustradamus** [comment]
<p>Hello <a class="mention" href="/darrowby385">@darrowby385</a>,</p>
<p>I wish you a Happy New Year!</p>
<p>Any news for this feature?</p>
<p>xmpp: links in markdown do not work, it is always mailto in file…</p>
<p>Examples:<br>
xmpp:example@domain.tld = xmpp:<a href="mailto:example@domain.tld">mailto:example@domain.tld</a><br>
xmpp:example@domain.tld?message = xmpp:<a href="mailto:example@domain.tld">mailto:example@domain.tld</a>?message<br>
xmpp:example@domain.tld?join = xmpp:<a href="mailto:example@domain.tld">mailto:example@domain.tld</a>?join</p>
<p>Can you see with team internally?</p>
<p>Thanks in advance.</p>

**Comment by canuckjacq** [comment]
<p>:wave:</p>
<p>I’m afraid this isn’t roadmapped at present. If you’re interested, we do have a public version of our roadmap here:</p>
<p><a href="https://github.com/github/roadmap" class="inline-onebox">GitHub - github/roadmap: GitHub public roadmap</a></p>

**Comment by Neustradamus** [comment]
<p><a class="mention" href="/canuckjacq">@canuckjacq</a>: Can you see to really add this feature?</p>
<p>It has been added in previous system by me: <a href="https://github.com/gjtorikian/html-pipeline/pull/307" class="inline-onebox" rel="noopener nofollow ugc">Add XMPP URI #289 by Neustradamus · Pull Request #307 · gjtorikian/html-pipeline · GitHub</a>.</p>
<p>There are a lot of tickets about it:</p>
<ul>
<li><a href="https://github.com/gjtorikian/html-pipeline/issues/289" class="inline-onebox" rel="noopener nofollow ugc">RFC5122: XMPP URI · Issue #289 · gjtorikian/html-pipeline · GitHub</a></li>
<li><a href="https://github.com/github/markup/issues/202" class="inline-onebox" rel="noopener nofollow ugc">Markdown parsing of URI schemes · Issue #202 · github/markup · GitHub</a></li>
<li><a href="https://github.com/github/markup/issues/426" class="inline-onebox" rel="noopener nofollow ugc">Markdown links don't seem to work if scheme is something "weird" · Issue #426 · github/markup · GitHub</a></li>
<li><a href="https://github.com/gjtorikian/html-pipeline/pull/178" class="inline-onebox" rel="noopener nofollow ugc">Add XMPP scheme by sonnyp · Pull Request #178 · gjtorikian/html-pipeline · GitHub</a></li>
</ul>
<p>Thanks in advance.</p>
